Source/SwiftLintBuiltInRules/Rules/Lint/ExpiringTodoRule.swift

```diff
@@ -45,73 +46,137 @@ struct ExpiringTodoRule: OptInRule {
-                return nil
+            // Process each comment individually
+            for token in node.tokens(viewMode: .sourceAccurate) {
+                processTrivia(
+                    token.leadingTrivia,
+                    baseOffset: token.position.utf8Offset,
+                    regex: regex
+                )
+                processTrivia(
+                    token.trailingTrivia,
+                    baseOffset: token.endPositionBeforeTrailingTrivia.utf8Offset,
+                    regex: regex
+                )
             }
 
-            return StyleViolation(
-                ruleDescription: Self.description,
-                severity: severity,
-                location: Location(file: file, characterOffset: range.location),
-                reason: violationLevel.reason
-            )
+            return .skipChildren
         }
-    }
 
-    private func expiryDate(file: SwiftLintFile, range: NSRange) -> Date? {
-        let expiryDateString = file.contents.bridge()
-            .substring(with: range)
-            .trimmingCharacters(in: .whitespacesAndNewlines)
+        private func processTrivia(_ trivia: Trivia, baseOffset: Int, regex: NSRegularExpression) {
+            var triviaOffset = baseOffset
 
-        let formatter = DateFormatter()
-        formatter.calendar = .current
-        formatter.dateFormat = configuration.dateFormat
+            for piece in trivia {
+                defer { triviaOffset += piece.sourceLength.utf8Length }
 
-        return formatter.date(from: expiryDateString)
-    }
+                guard let commentText = piece.commentText else { continue }
 
-    private func severity(for violationLevel: ExpiryViolationLevel) -> ViolationSeverity? {
-        switch violationLevel {
-        case .approachingExpiry:
-            return configuration.approachingExpirySeverity.severity
-        case .expired:
-            return configuration.expiredSeverity.severity
-        case .badFormatting:
-            return configuration.badFormattingSeverity.severity
+                // Handle multiline comments by checking consecutive line comments
+                if piece.isLineComment {
+                    var combinedText = commentText
+                    let currentOffset = triviaOffset
+
+                    // Look ahead for consecutive line comments
+                    let remainingTrivia = trivia.dropFirst(trivia.firstIndex(of: piece)! + 1)
+
+                    for nextPiece in remainingTrivia {
+                        if case .lineComment(let nextText) = nextPiece {
+                            // Check if it's a continuation (starts with //)
+                            if nextText.hasPrefix("//") {
+                                combinedText += "\n" + nextText
+                            } else {
+                                break
+                            }
+                        } else if !nextPiece.isNewline && !nextPiece.isWhitespace {
```

Comment:
the built-in `isWhiteSpace` covers spaces, tabs, and newlines.

```suggestion
                        } else if !nextPiece.isWhitespace {
```

Then delete the two computed vars at the bottom of the file.